File: frontend/src/app/models/optionList.ts
import {User} from "./user";
import {OptionValue} from "./optionValue";
import {Question} from "./question";

export class OptionList {

    id: number;
    name: string;
    owner?: User;
    listOptionValues?: OptionValue[];

    constructor(data:any) {
        this.id = data.id;
        this.name = data.name;
        this.owner = data.owner;
        if (data.listOptionValues) {
            this.listOptionValues = data.listOptionValues.map((value: any) => new OptionValue(value));
        }
    }
}

File: backend/Models/OptionList/OptionList.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace prid_2425_a01.Models.OptionList;


public class OptionList {
    [Key]
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public int? OwnerId { get; set; } 

    [ForeignKey(nameof(OwnerId))]
    public User.User Owner { get; set; }= null!;
    

    public ICollection<OptionValue.OptionValue> ListOptionValues { get; set; } = new HashSet<OptionValue.OptionValue>();
}
